### ISSUE TITLE
Is it possible to deploy opentelemetry-operator without clusterrole, using role instead

### ISSUE THREAD

**Issue by chunglun** [open-telemetry/opentelemetry-operator#3042]
### Describe the issue you're reporting

Is it possible to deploy opentelemetry-operator without clusterrole, using role instead?
My company doesn't want to grant me cluster admin privilege and ask me to use role and rolebing only to deploy this operator; however, first error happened 

go@v0.29.3/tools/cache/reflector.go:229: failed to list *v1.PersistentVolume: persistentvolumes is forbidden: User "system:serviceaccount:..." cannot list resource "persistentvolumes" in API group "" at the cluster scope

How could I solve it.....

**Comment by swiatekm-sumo** [comment]
You mean for the operator itself? You should be able to, but as a Role only covers RBAC in a specific namespace, the operator deployed this way will only be able to affect resource in this namespace. You'll want to set the WATCH_NAMESPACE env variable to that namespace as well.

**Comment by pavolloffay** [comment]
Let us know if that worked. I have seen this request before and it would be great to document it.

**Comment by chunglun** [comment]
In fact, I moved most of the setting in ClusterRole to Role but keep the get/list/watch persistentVolume in ClusterRole.

Operator was successfully deployed and Center Collector was successfully deployed too.

Currently I’m solving the problem of auto instrumentation due to Instrument deployed but no pods generates. Not sure what makes auto instrumentation not work, still work on it.

**Comment by iblancasa** [comment]
> Currently I’m solving the problem of auto instrumentation due to Instrument deployed but no pods generated. Not sure what makes auto instrumentation not work, still work on it.

The `Instrumentation` CR doesn't deploy any pod. It creates an `InitContainer` in those pods where you annotate to inject the autoinstrumentation.

**Comment by chunglun** [comment]
> > Currently I’m solving the problem of auto instrumentation due to Instrument deployed but no pods generated. Not sure what makes auto instrumentation not work, still work on it.
> 
> The `Instrumentation` CR doesn't deploy any pod. It creates an `InitContainer` in those pods where you annotate to inject the autoinstrumentation.

!!! I will try again tomorrow to see if there is an InitContainer. Thanks so much for this helpful information.

**Comment by alita1991** [comment]
Hi, I have the same scenario to cover and encountered the same issues when I tried to bootstrap some collectors via the operator. Probably I need to create a new role with enough access or I miss something else that I'm not aware of. Right now, the operator is configured to watch only a specific namespace, where the operator is deployed.

**Comment by alita1991** [comment]
After I spent a little bit of time, I found out that even though the operator is configured to watch namespace(s), there is some logic in the code, where the manager is doing a watch on `PersistentVolume` resources, which are cluster-wide. 

```
func (r *OpenTelemetryCollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
	builder := ctrl.NewControllerManagedBy(mgr).
		For(&v1beta1.OpenTelemetryCollector{}).
		Owns(&corev1.ConfigMap{}).
		Owns(&corev1.ServiceAccount{}).
		Owns(&corev1.Service{}).
		Owns(&appsv1.Deployment{}).
		Owns(&appsv1.DaemonSet{}).
		Owns(&appsv1.StatefulSet{}).
                 // Owns(&corev1.PersistentVolume{}). <-- I comment this line
		Owns(&corev1.PersistentVolumeClaim{}).
		Owns(&networkingv1.Ingress{}).
		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
		Owns(&policyV1.PodDisruptionBudget{})
```

After I removed that specific line, the operator had no runtime issues. Because I never did any development on the otel-operator, I do not know the impact, any feedback/solution would be appreciated.

**Comment by pavolloffay** [comment]
If a single namespace is watched the cluster scope objects could be excluded. But it does seem that it would change the operator's feature set.

**Comment by chunglun** [comment]
> In fact, I moved most of the setting in ClusterRole to Role but keep the get/list/watch persistentVolume in ClusterRole and remove delete/create/update/patch persistentVolume due to privilege limitation in my company.
> 
> Operator was successfully deployed and Center Collector was successfully deployed too.
> 
> Currently I’m solving the problem of auto instrumentation due to Instrument deployed but no pods generated. Not sure what makes auto instrumentation not work, still work on it.

In fact, I still fail on the auto instrumentation, no any initial container appeared on my application. I kinds of want to drop the operator, use the combination of plain yaml collector deployment + ingest the cleint agent jar to application vai initial comtainer setting in application's deployment yaml to let the application fulfill no any rebuild applciation image needed

**Comment by swiatekm** [comment]
> After I spent a little bit of time, I found out that even though the operator is configured to watch namespace(s), there is some logic in the code, where the manager is doing a watch on `PersistentVolume` resources, which are cluster-wide.
> 
> ```
> func (r *OpenTelemetryCollectorReconciler) SetupWithManager(mgr ctrl.Manager) error {
> 	builder := ctrl.NewControllerManagedBy(mgr).
> 		For(&v1beta1.OpenTelemetryCollector{}).
> 		Owns(&corev1.ConfigMap{}).
> 		Owns(&corev1.ServiceAccount{}).
> 		Owns(&corev1.Service{}).
> 		Owns(&appsv1.Deployment{}).
> 		Owns(&appsv1.DaemonSet{}).
> 		Owns(&appsv1.StatefulSet{}).
>                  // Owns(&corev1.PersistentVolume{}). <-- I comment this line
> 		Owns(&corev1.PersistentVolumeClaim{}).
> 		Owns(&networkingv1.Ingress{}).
> 		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
> 		Owns(&policyV1.PodDisruptionBudget{})
> ```
> 
> After I removed that specific line, the operator had no runtime issues. Because I never did any development on the otel-operator, I do not know the impact, any feedback/solution would be highly appreciated.

This is very puzzling to me. There isn't any difference between `PersistentVolume` and other namespaced resources on that list, and I can't see any reason why it specifically would lead to a problem.

On a separate note, @pavolloffay @jaronoff97 do you know why we try to own Volumes and Claims here? I don't think we create any.

**Comment by jaronoff97** [comment]
good question... initially I thought it was related to statefulsets, but any PVs and PVCs would be under the statefulset management tree. I don't think there's a reason we need to keep that in there...

**Comment by pavolloffay** [comment]
Yes, it's perhaps bc of the statefulset. +1 on removing it if it is not required.
